mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,415 @@
+on every batch *training* or *evaluation* was done.
+
+### Implementation details
+
+#### Query module
+Module is implemented using **[PyTorch](https://pytorch.org/)**. From `mgp.Edge` input features defined as edge label, 
+we extract `edge features` and from `mgp.Vertex` input features we extract `node features`. Since we expect in *your* learning activities 
+to use **triggers**, the `update` query module procedure will parse all new edges, extract information **temporal graph networks** need to do a 
+batch by batch processing.
+
+From following piece of code, *you* can see what we extract from every edge, while filling up **batch**. When our current processing
+batch size reaches **predefined batch size**, predefined by calling `set` query module procedure, we `forward` extracted information to `TGN` which
+extends `torch.nn.Module`.
+```python
+@dataclasses.dataclass
+class QueryModuleTGNBatch:
+    current_batch_size: int
+    sources: np.array
+    destinations: np.array
+    timestamps: np.array
+    edge_idxs: np.array
+    node_features: Dict[int, torch.Tensor]
+    edge_features: Dict[int, torch.Tensor]
+    batch_size: int
+    labels: np.array
+
+
+```
+
+#### Processing one batch
+
+
+```python
+        self._process_previous_batches()
+
+        graph_data = self._get_graph_data(
+            np.concatenate([sources.copy(), destinations.copy()], dtype=int),
+            np.concatenate([timestamps, timestamps]),
+        )
+
+        embeddings = self.tgn_net(graph_data)
+
+        ... process negative edges in a similar way
+        
+        self._process_current_batch(
+            sources, destinations, node_features, edge_features, edge_idxs, timestamps
+        )
+
+```
+Our `torch.nn.Module` is organized as follows:
+  * processing previous batches - if you follow *[research paper](https://arxiv.org/abs/2006.10637)* this will include a new calculation of messages collected for each node in the form of 
+    **message function**, aggregation of messages for each node in form of **message aggregator** and finally updating of each of the node's memory
+    with **memory updater**
+  * afterward we create a computation graph used by **graph attention layer** or **graph sum layer**
+  * final step includes processing of current batch, creating new **interaction or node events**, updating **raw message store** with new **events**
+
+Process repeats as we get new edges in batch, and after a batch is filled, we forward new edges again to **TGN** and so on...
+
+
+
+:::info
+
+This **MAGE** module is still in its early phase. We intended its use for only **learning** activities. The problem 
+with the current module is that we expect **you** to know when your stream will end so you can switch **TGN** mode
+from **train** to **eval**. After you do a switch, all incoming edges will be used for only **evaluation** and not
```

Comment:
I would phrase this in the terms of the module only supporting learning so far as evaluation needs to be switched on manually.

Reply:
No, this is okay in my opinion. We need to tell users what is implemented and what can be used.